XenAdmin/Diagnostics/Checks/UpgradeRequiresEUA.cs

```diff
@@ -70,7 +73,7 @@ public UpgradeRequiresEua(Control control, List<Host> hosts, IReadOnlyDictionary
             _hostsFailedToFetchEua = new HashSet<IXenObject>();
         }
 
-        public override bool CanRun() => _targetUri != null && Hosts.Any(Helpers.YangtzeOrGreater);
+        public override bool CanRun() => _targetUri != null;

```

Comment:
I noticed two more issues, the code is in fact in RollingUpgradeWizardPrecheckPage.cs
```
if (!ManualUpgrade)
{
    // EUA check
    var euaCheck = GetPermanentCheck("EUA",
        new UpgradeRequiresEua(this, hostsToUpgrade, InstallMethodConfig));
    var euaChecks = new List<Check> { euaCheck };
    groups.Add(new CheckGroup(Messages.ACCEPT_EUA_CHECK_GROUP_NAME, euaChecks));

    var safeToUpgradeChecks = (from Host host in hostsToUpgrade
        let check = new SafeToUpgradeCheck(host, InstallMethodConfig)
        where check.CanRun()
        select check as Check).ToList();

    if (safeToUpgradeChecks.Count > 0)
        groups.Add(new CheckGroup(Messages.CHECKING_SAFE_TO_UPGRADE, safeToUpgradeChecks));

    var prepareToUpgradeChecks = (from Host host in hostsToUpgrade
        select new PrepareToUpgradeCheck(host, InstallMethodConfig) as Check).ToList();

    if (prepareToUpgradeChecks.Count > 0)
        groups.Add(new CheckGroup(Messages.CHECKING_PREPARE_TO_UPGRADE, prepareToUpgradeChecks));
}
```

- In the above code where we construct the UpgradeRequiresEua we don't check if it CanRun(), so it runs successfully even if the Uri is invalid (because the constructor just logs the error quietly in this case).
- This is an older issue: I find a bit strange that the PrepareToUpgradeCheck, which checks for invalid installer URLs, comes last  in this block. I think it should be first.

Reply:
> In the above code where we construct the UpgradeRequiresEua we don't check if it CanRun(), so it runs successfully even if the Uri is invalid (because the constructor just logs the error quietly in this case).

Actually `CanRun` runs as part of `RunAllChecks()`, which is what actually runs `RunCheck()`.

However, there is a commment I missed that points out that the check should not be added at all if `CanRun` isn't valid, so this needs changing anyway

> This is an older issue: I find a bit strange that the PrepareToUpgradeCheck, which checks for invalid installer URLs, comes last in this block. I think it should be first.

sure! 👍